### DISCUSSION TITLE
Affected by PyCryptodome vulnerability (CVE-2023-52323)

### DISCUSSION THREAD

**Discussion by Razortide** [pysnmp/pysnmp#73] [General]
Hello all,

we noticed that https://nvd.nist.gov/vuln/detail/CVE-2023-52323 is part of the pysnmp dependencies. As far as we can tell, the PyCryptodome version i3.10.1 s only used in asynchronous and not in the synchronous encryption in pysnmp v4.4.12-1.el8 (see https://github.com/pysnmp/pysnmp). Can you please confirm that this is the case? Since we are using EPEL8, we don't have a possibility to upgrade easily.

Best regards,
Marc

**Comment by lextm** [comment]
1. Ilya foresaw such and fixed in his 5.0 release branch. However, he wasn't able to ship the changes in any 4.4.x releases.
2. This fork `pysnmplib` doesn't contain the changes either.
3. `pysnmp-lextudio` 5.0.27+ includes Ilya's changes, so the dependency can be changed from PyCryptodomex to pyca/cryptography.

https://github.com/lextudio/pysnmp/blob/main/CHANGES.rst#revision-5027-released-on-apr-28-2023

I don't think "the PyCryptodome version i3.10.1 s only used in asynchronous and not in the synchronous encryption in pysnmp v4.4.12-1.el8" is correct. All crypo related operations in 4.4.12 go through PyCryptodome, sync or async.
